source/ranges.tex

```diff
@@ -302,170 +325,191 @@
+  namespace views {
+    inline constexpr @\unspec@ slide = @\unspec@;                               // freestanding
+  }
 
   // \ref{range.chunk.by}, chunk by view
   template<@\libconcept{forward_range}@ V, @\libconcept{indirect_binary_predicate}@<iterator_t<V>, iterator_t<V>> Pred>
     requires @\libconcept{view}@<V> && is_object_v<Pred>
-  class chunk_by_view;
+  class chunk_by_view;                                                              // freestanding
 
-  namespace views { inline constexpr @\unspec@ chunk_by = @\unspec@; }
+  namespace views {
+    inline constexpr @\unspec@ chunk_by = @\unspec@;                            // freestanding
+  }
 }
 
 namespace std {
   namespace views = ranges::views;
 
-  template<class T> struct tuple_size;
-  template<size_t I, class T> struct tuple_element;
+  template<class T> struct tuple_size;                                              // freestanding
+  template<size_t I, class T> struct tuple_element;                                 // freestanding
 
   template<class I, class S, ranges::subrange_kind K>
-  struct tuple_size<ranges::subrange<I, S, K>>
+  struct tuple_size<ranges::subrange<I, S, K>>                                      // freestanding
     : integral_constant<size_t, 2> {};
   template<class I, class S, ranges::subrange_kind K>
-  struct tuple_element<0, ranges::subrange<I, S, K>> {
-    using type = I;
+  struct tuple_element<0, ranges::subrange<I, S, K>> {                              // freestanding
+    using type = I;                                                                 // freestanding
   };
   template<class I, class S, ranges::subrange_kind K>
-  struct tuple_element<1, ranges::subrange<I, S, K>> {
-    using type = S;
+  struct tuple_element<1, ranges::subrange<I, S, K>> {                              // freestanding
+    using type = S;                                                                 // freestanding
   };
   template<class I, class S, ranges::subrange_kind K>
-  struct tuple_element<0, const ranges::subrange<I, S, K>> {
-    using type = I;
+  struct tuple_element<0, const ranges::subrange<I, S, K>> {                        // freestanding
+    using type = I;                                                                 // freestanding
   };
   template<class I, class S, ranges::subrange_kind K>
-  struct tuple_element<1, const ranges::subrange<I, S, K>> {
-    using type = S;
+  struct tuple_element<1, const ranges::subrange<I, S, K>> {                        // freestanding
+    using type = S;                                                                 // freestanding
   };
 
-  struct from_range_t { explicit from_range_t() = default; };
-  inline constexpr from_range_t from_range{};
+  struct from_range_t {                                                             // freestanding
+    explicit from_range_t() = default;                                              // freestanding
```

Comment:
I found it odd that, at the end of `<ranges>`, you went out of your way to add append `// `_`freestanding`_ to members. The paper does indeed instruct you to do this with
>  Please append a `// `_`freestanding`_ comment to every entity in `<`...`>` except for the following entities:

I wonder if this is necessary at all. [freestanding.entity]p2 seems to cover this:
> Unless otherwise specified, the requirements on freestanding entities on a freestanding implementation are the same as the corresponding requirements in a hosted implementation.

Otherwise, I wonder why not do the same to the class synopsis of all freestanding entities.

Reply:
In that regards, you missed adding `// `_`freestanding`_ to members of the following entities:
`allocator_arg_t`, `allocation_result`, and `initializer_list` (unless it gets its own actual header synopsis, see https://github.com/cplusplus/draft/pull/5713/files#r939552729).

Reply:
In which header synopsis did I miss these? I don't see them in ranges

Reply:
They're in these files:
```terminal
$ grep class\ initializer_list *.tex
support.tex:  template<class E> class initializer_list {
$ grep struct\ allocator_arg_t *.tex
memory.tex:  struct allocator_arg_t { explicit allocator_arg_t() = default; };
```
I think you can ignore `initializer_list` due to https://github.com/cplusplus/draft/pull/5713#discussion_r939552729. https://github.com/cplusplus/draft/pull/5713#discussion_r939554688 applies to this, too, so you can keep it on a single line.

Reply:
`class initializer_list` in [initializer.list.syn] doesn't need to be marked because the header is marked `all freestanding`

fixed `struct allocator_arg_t` in [memory.syn], thanks

assuming you're talking about the members of `struct allocation_result` in [memory.syn], they were not listed among the entities to be marked in the paper

So I think we're good now, right?

Reply:
That's right. Though I wonder why the class members of `allocator_arg_t` and of the specializations at the end of [ranges.syn] need to be tagged as freestanding but not those of `allocation_result`.